endgame_postprocessing/model_wrappers/sch/run_sch.py

```diff
@@ -193,56 +193,38 @@ def _check_iu_in_all_folders(worm_iu_info):
 def canonicalise_raw_sch_results(
     input_dir,
     output_dir,
-    all_worm=False,
-    worm_directories = [],
-    first_worm=""
+    worm_directories,
 ):
-    if all_worm:
-        if len(worm_directories) <= 0:
-            raise Exception(
-                "Need to supply a list of worm directories for calculating all worm for SCH"
-            )
-        if first_worm == "" or first_worm not in worm_directories:
-            raise Exception(
-                "First worm needs to be supplied and in the worm directories"
-            )
-    if not all_worm:
-        if len(worm_directories) > 0 or first_worm != "":
-            raise Warning(
-                "Parameters `worm_directories` and `first_worm` were provided" +
-                ", but are unused when all_worm is set to False"
-            )
-
+    if len(worm_directories) < 1:
+        raise Exception(
+            "Expected at least 1 item in the worm_directories parameter," +
+            f"received {len(worm_directories)}."
+        )
     # Assuming that the first worm only has one burden
-    file_iter_addition = ""
-    other_worms = []
-    if all_worm:
-        file_iter_addition = first_worm
-
-    file_iter = get_sch_flat(f"{input_dir}{file_iter_addition}")
-
-    if all_worm:
-        other_worm_directories = [worm for worm in worm_directories if worm != first_worm]
-        iter_other_worms = [
-            list(get_sch_flat(f"{input_dir}{other_worm_dir}"))
-            for other_worm_dir in other_worm_directories
-        ]
-        flattened_iter_other_worms = [
-            file.file_path
-            for worm_list in iter_other_worms
-            for file in worm_list
-        ]
-        other_worms = [
-            get_sch_worm_info(next(get_sch_flat(f"{input_dir}{other_worm_dir}")).file_path)
-            for other_worm_dir in other_worm_directories
-        ]
+    first_worm = worm_directories[0]
+    file_iter = get_sch_flat(f"{input_dir}{first_worm}")
 
-        all_iu_worm_info = [
-            get_sch_worm_info(file_info.file_path)
-            for worm_dir in worm_directories
-            for file_info in get_sch_flat(f"{input_dir}{worm_dir}")
-        ]
-        _check_iu_in_all_folders(all_iu_worm_info)
+    other_worm_directories = worm_directories[1:]
+    iter_other_worms = [
+        list(get_sch_flat(f"{input_dir}{other_worm_dir}"))
+        for other_worm_dir in other_worm_directories
+    ]
+    flattened_iter_other_worms = [
+        file.file_path
+        for worm_list in iter_other_worms
+        for file in worm_list
+    ]
```

Comment:
About the two above statements:
- I would move them closer to the line `flattened_iter_other_worms` is actually used, which is quite further down at line 249.
- I would also make use of the fact that `get_sch_flat`returns a generator function to avoid converting to a list and compute a set of all the file paths directly:
    ```python
    all_other_worms_file_paths = set(
        file_desc.file_path
        for worm_files_list in (
            get_sch_flat(f"{input_dir}{worm_dir}")
            for worm_dir in other_worm_directories
         )
         for file_desc in worm_files_list
    )
    for worm, burden, _, _ in other_worms:
        # ...
        if new_file.file_path in all_other_worms_file_paths:
            other_worm_file_infos.append(new_file)
    ```

In addition I renamed `file` to `file_desc` since `file` is a reserved Python keyword.